modules/apps/portal-template/portal-template-react-renderer-impl/src/main/java/com/liferay/portal/template/react/renderer/internal/ReactRendererUtil.java

```diff
@@ -119,6 +120,14 @@ private static void _renderJavaScript(
 		dependenciesSB.append(" as renderFunction");
 		dependenciesSB.append(placeholderId);
 
+		Set<String> componentDependencies = componentDescriptor.getDependencies();
+
+		for (String componentDependency : componentDependencies) {
+			dependenciesSB.append(", ");
+			dependenciesSB.append(componentDependency);
+			dependenciesSB.append(" as componentDep");
```

Comment:
Can this potentially cause multiple variables with the same name to be defined?

Reply:
See above... 

> Disregard most of the PR code-wise since it's pretty fuggly... looking for general feedback as to where to take this... 🤔

Reply:
And yes, this I wasn't planning on keeping ;)

Reply:
Gotcha

Reply:
Outdated, added a new dependency if we have a propsTransformer -> https://github.com/liferay-frontend/liferay-portal/pull/2/files#diff-6342875c29a4e8bea3611cce5b3f9d62R126